### PR TITLE
[IMP] add addons_path config option

### PR DESCRIPTION
**Pull request by blaggacao**
addons_path is, after database, the most volatile configuration parameter
only known at runtime in many cases (contrary to other config file items).

Expose this configuration parameter appropriately.

**Note:** See how the addons folder is construed dynamically in [this use case](https://github.com/xoe-labs/dockery-odoo/blob/80cb0647ccff285c4450c9f4a8410e7bfc298ec5/images/base/v-11.0/out/entrypoint.d/10-set-appenv.sh#L18-L32).

### CODE REVIEW

File: click_odoo/cli.py [hunk 1]
```diff
@@ -31,6 +31,10 @@ def inner(func):
                            "OPENERP_SERVER environment variables, "
                            "or ~/.odoorc (Odoo >= 10) "
                            "or ~/.openerp_serverrc.")
+        @click.option('--addons-path', envvar=['ODOO_ADDONSPATH'],
+                      help="Specify the addons path. If present, this "
+                           "parameter takes precedence over the database "
```
Comment (sbidoul) [inline click_odoo/cli.py]: s/database/addons path/

File: click_odoo/cli.py [hunk 2]
```diff
@@ -31,6 +31,10 @@ def inner(func):
                            "OPENERP_SERVER environment variables, "
                            "or ~/.odoorc (Odoo >= 10) "
                            "or ~/.openerp_serverrc.")
+        @click.option('--addons-path', envvar=['ODOO_ADDONSPATH'],
```
Comment (sbidoul) [inline click_odoo/cli.py]: s/ODOO_ADDONSPATH/ODOO_ADDONS_PATH/, because :)

**Comment by sbidoul** [comment]
And update the README as well, thanks!

File: click_odoo/cli.py (hunk 2 above)
Comment (blaggacao) [reply click_odoo/cli.py]: :wink:

**Comment by blaggacao** [comment]
@sbidoul Mostly done, test coverage will come later today... Thanks!

File: README.rst [hunk 3]
```diff
@@ -149,18 +149,22 @@ Command line interface (click-odoo)
     to be a terminal.
 
   Options:
-    -c, --config PATH               Specify the Odoo configuration file. Other
+    -c, --config FILE               Specify the Odoo configuration file. Other
```
Comment (sbidoul) [inline README.rst]: do not change this, it is what click-odoo --help prints.

Comment (sbidoul) [reply README.rst]: ah, no, sorry, click 7.0 changed that, good!

**Comment by blaggacao** [comment]
@sbidoul All green :smiley: I guess it's ready?

File: tests/test_click_odoo.py [hunk 4]
```diff
@@ -235,6 +235,55 @@ def testcmd(env):
...
+        db_name=notadb
+        addons_path=notanaddonspath
+    """))
+    result = runner.invoke(testcmd, [
+        '-c', str(odoocfg2),
+        '-d', odoodb,
+        '--addons-path', addons_path,
+    ])
+    assert result.exit_code == 0
+    assert 'login=admin\n' in result.output
```
Comment (sbidoul) [inline tests/test_click_odoo.py]: This test does not seem to test the addons path is actually taken into account.
There is no need to do a `search` in this test, nor do we need the `odoodb` fixture.

I suggest writing a `script5.py` which simply does `import odoo.addons.addon1` and call it using subprocess (`click-odoo -- script5.py`) in the test to check if it succeeds or not depending on the addons path (see other subprocess tests).

Comment (blaggacao) [reply tests/test_click_odoo.py]: Much better :+1:

File: tests/test_click_odoo.py [hunk 5]
```diff
@@ -235,6 +235,24 @@ def testcmd(env):
     assert "No database provided" in result.output
 
 
+def test_env_options_addons_path():
+    addons_path = ','.join([
+        os.path.join(odoo.__path__[0], 'addons'),
+        os.path.join(odoo.__path__[0], '..', 'addons'),
```
Comment (blaggacao) [inline tests/test_click_odoo.py]: @sbidoul Couldn't this line be also dropped?

Comment (sbidoul) [reply tests/test_click_odoo.py]: I think so, yes

Comment (blaggacao) [reply tests/test_click_odoo.py]: Ok, removed.

**Comment by blaggacao** [comment]
@sbidoul Finally :smile:

File: tests/scripts/script5.py [hunk 6]
```diff
@@ -0,0 +1,6 @@
+
+try:
+    import odoo.addons.addon1  # noqa
+except ImportError:
+    #  Odoo < 10.0
+    import openerp.addons.addon1  # noqa
```
Comment (sbidoul) [inline tests/scripts/script5.py]: Hm, this is not correct as the absence of addon1 in addons-path would raise the wrong error. Ideally there should be a test that checks script5 fails if addon1 is not in addons-path.

And this script should be

```python
# noqa
try:
    import odoo
except ImportError:
    #  Odoo < 10.0
    try:
        import openerp
    except ImportError:
        # odoo not found
        raise
    else:
        import openerp.addons.addon1
else:
    import odoo.addons.addon1
```

File: tests/test_click_odoo.py [hunk 7]
```diff
@@ -235,6 +235,23 @@ def testcmd(env):
     assert "No database provided" in result.output
 
 
+def test_env_options_addons_path():
```
Comment (sbidoul) [inline tests/test_click_odoo.py]: So here, start checking script5 fails when addons-path is not set

File: tests/scripts/script5.py (hunk 6 above)
Comment (blaggacao) [reply tests/scripts/script5.py]: Oh thanks!

File: CHANGES.rst [hunk 8]
```diff
@@ -5,6 +5,12 @@ Changes
 .. ----------
 .. - ...
 
+1.0.5 (2018-10-26)
+------------------
+- add --addons-path parameter as a very volatile config option
```
[the remark this answers is not in the document]
Comment (blaggacao) [reply CHANGES.rst]: Done

File: tests/test_click_odoo.py (hunk 7 above)
Comment (blaggacao) [reply tests/test_click_odoo.py]: done

**Comment by blaggacao** [comment]
@sbidoul There is some problem with the odoo namespace in version 9.0/8.0 I don't get it. :cry:

**Comment by sbidoul** [comment]
Merged manually at 33f2b1f777f1a4cf6e66979e51e10cc9c24f8bcb and improved test at 3cb4c7bbbe39c0d72b14cfa0049e9f2eaaa0b06a. Thanks!

**Comment by blaggacao** [comment]
Thank you a lot @sbidoul, I was really struggling with those tests... :wink:

**Pull request closed without merging**